src/main/java/net/minestom/server/utils/FinalObject.java

```diff
@@ -0,0 +1,48 @@
+package net.minestom.server.utils;
+
+import java.util.function.Supplier;
+
+/**
+ * Used to mimic the final keyword
```

Comment:
Well that's not really true, it's more of a lazy initialized object than a final object, i.e. it can only be set once, and can be set lazily.

Reply:
Can I recommend a new name like `LazyInitializedObject` or `LazyLoadedValue` for this class?

Reply:
We try to avoid this object, so it's name probably won't matter, but if it ends up in the final pr i'll choose a better name.